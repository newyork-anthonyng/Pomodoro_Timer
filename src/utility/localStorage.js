const Storage = (function() {
  const KEY = 'pomodoro_timer';

  const get = () => {
    try {
      return JSON.parse(window.localStorage.getItem(KEY));
    } catch (e) {
      return [];
    }
  };

  const set = (favorites) => {
    window.localStorage.setItem(KEY, JSON.stringify(favorites));
  };

  return {
    get,
    set,
  };
})();

export default Storage;
