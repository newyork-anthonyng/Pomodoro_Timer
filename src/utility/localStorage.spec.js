import localStorage from './localStorage';

beforeEach(() => {
  window.localStorage = {
    getItem: jest.fn(() => '[{ "foo": "bar" }]'),
    setItem: jest.fn(),
  };
});

describe('localStorage.get', () => {
  it('should get items from localStorage', () => {
    const result = localStorage.get();

    expect(window.localStorage.getItem).toHaveBeenCalledTimes(1);
    expect(window.localStorage.getItem.mock.calls[0]).toMatchSnapshot();
    expect(result).toMatchSnapshot();
  });

  it('should return empty array if there is an error', () => {
    window.localStorage.getItem = jest.fn(() => {
      return '{INVALID_JSON';
    });
    let result;

    expect(() => {
      result = localStorage.get();
    }).not.toThrow();
    expect(result).toMatchSnapshot();
  });
});

describe('localStorage.updateTasks', () => {
  it('should set items in localStorage', () => {
    localStorage.set([{ foo: 'bar' }]);

    expect(window.localStorage.setItem).toHaveBeenCalledTimes(1);
    expect(window.localStorage.setItem.mock.calls[0]).toMatchSnapshot();
  });
});
