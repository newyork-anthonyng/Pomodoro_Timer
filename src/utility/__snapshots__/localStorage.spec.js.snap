// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`localStorage.get should get items from localStorage 1`] = `
Array [
  "pomodoro_timer",
]
`;

exports[`localStorage.get should get items from localStorage 2`] = `
Array [
  Object {
    "foo": "bar",
  },
]
`;

exports[`localStorage.get should return empty array if there is an error 1`] = `Array []`;

exports[`localStorage.updateTasks should set items in localStorage 1`] = `
Array [
  "pomodoro_timer",
  "[{\\"foo\\":\\"bar\\"}]",
]
`;
